iis-wf/iis-wf-export-actionmanager/src/main/java/eu/dnetlib/iis/wf/export/actionmanager/entity/AbstractEntityExporterProcess.java

```diff
@@ -43,225 +41,219 @@
+        this.entityXSLTName = entityXSLTName;
+        this.entityXSLTLocation = entityXSLTLocation;
+        this.entityNamespacePrefix = entityNamespacePrefix;
+        this.actionFactory = buildActionFactory();
+
+    }
+
+    // ------------------------ LOGIC -----------------------------
+
+    @Override
+    public void run(PortBindings portBindings, Configuration conf, Map<String, String> parameters) throws Exception {
+        String actionSetId = ProcessUtils.getParameterValue(EXPORT_ACTION_SETID, conf, parameters);
+        Preconditions.checkArgument(StringUtils.isNotBlank(actionSetId) && !WorkflowRuntimeParameters.UNDEFINED_NONEMPTY_VALUE.equals(actionSetId),
+                "unable to export document entities to action manager due to missing action set identifier, "
+                        + "no '%s' required parameter provided!", EXPORT_ACTION_SETID);
+
+        FileSystemPath inputPath = new FileSystemPath(FileSystem.get(conf), portBindings.getInput().get(inputPort));
+
+        try (ActionManagerServiceFacade actionManager = buildActionManager(conf, parameters);
+                CloseableIterator<T> it = DataStore.<T> getReader(inputPath)) {
+            
+            MDStoreFacade mdStore = buildMDStoreFacade(parameters);
+
+            int counter = 0;
+            while (it.hasNext()) {
+                MDStoreIdWithEntityId mdStoreComplexId = deliverMDStoreWithEntityId(it.next());
+                String mdRecordId = convertToMDStoreEntityId(mdStoreComplexId.getEntityId());
+                try {
+                    String mdRecord = mdStore.deliverRecord(mdStoreComplexId.getMdStoreId(), mdRecordId);
+                    handleRecord(mdRecord, actionSetId, actionManager);
+                    counter++;
+                } catch (DocumentNotFoundException e) {
+                    log.error("mdrecord: " + mdRecordId + " wasn't found in mdstore: "
+                            + mdStoreComplexId.getMdStoreId(), e);
+                } catch (Exception e) {
+                    log.error("got exception when trying to retrieve " + "MDStore record for mdstore id "
+                            + mdStoreComplexId.getMdStoreId() + ", and document id: " + mdRecordId, e);
+                    throw e;
+                }
+
+            }
+            log.warn("exported " + counter + " entities in total");
+        }
+    }
+
+    @Override
+    public Map<String, PortType> getInputPorts() {
+        HashMap<String, PortType> inputPorts = Maps.newHashMap();
+        inputPorts.put(inputPort, new AvroPortType(inputPortSchema));
+        return inputPorts;
+    }
+
+    @Override
+    public Map<String, PortType> getOutputPorts() {
+        return Collections.emptyMap();
+    }
+
+    /**
+     * Provides mdstore and entity identifiers for given input record.
+     * 
+     * @return mdstore and entity identifiers pair
+     */
+    abstract protected MDStoreIdWithEntityId deliverMDStoreWithEntityId(T element);
```

Comment:
Deliver is a synonym of 'give', 'pass', 'sent' etc. Is it a proper verb for this method? Doesn't it just create/ convert the given element to the MDStoreIdWithEntityId. Deliver suggests the element is sent/ saved somewhere, but it is not.